chrononaut/flask_versioning.py

```diff
@@ -2,100 +2,171 @@
 """
 from flask import g
 from flask.globals import _app_ctx_stack
+from datetime import datetime
+from dateutil.tz import tzutc
+import six
+import numbers
 
-from sqlalchemy.orm import attributes, object_mapper
+from sqlalchemy import inspect
+from sqlalchemy.orm import object_mapper
 from sqlalchemy.orm.exc import UnmappedColumnError
-from sqlalchemy.orm.properties import RelationshipProperty
 
 from chrononaut.exceptions import ChrononautException
+from chrononaut.models import HistorySnapshot
 
 
-def fetch_change_info(obj):
-    change_info = obj._capture_change_info()
-    if _app_ctx_stack.top is None:
-        return change_info
-
-    extra_change_info = {}
-    extra_change_info.update(getattr(g, "__version_extra_change_info__", {}))
-    extra_change_info.update(getattr(obj, "__CHRONONAUT_RECORDED_CHANGES__", {}))
-    if extra_change_info:
-        change_info["extra"] = extra_change_info
-
-    return change_info
+UTC = tzutc()
 
 
-def create_version(obj, session, deleted=False):
-    obj_mapper = object_mapper(obj)
-    history_mapper = obj.__history_mapper__
-    history_cls = history_mapper.class_
+def serialize_datetime(dt):
+    try:
+        return dt.astimezone(UTC).replace(tzinfo=None).isoformat() + "+00:00"
+    except TypeError:
```

Comment:
when would we get a type error?

Reply:
We'll get a type error if we get a timezone unaware datetime. The latter is rare, but does happen, hence handling it as an exception.